addons/website/static/src/snippets/s_embed_code/000.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+import publicWidget from "web.public.widget";
+import { _t } from "@web/core/l10n/translation";
+
+const EmbedCodeWidget = publicWidget.Widget.extend({
+    selector: ".s_embed_code",
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    async start() {
+        this.embedCodeEl = this.el.querySelector(".s_embed_code_embedded");
+        if (this.editableMode && this.embedCodeEl.offsetHeight === 0) {
+            this._addPlaceholder();
+        }
+        return await this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        const embedContent = this.embedCodeEl.dataset.embedContent;
+        if (!this.editableMode && embedContent && embedContent.match(/<script[^<]*?>/)) {
+            this.embedCodeEl.innerHTML = embedContent.replace(/<script[^<]*?>.*?<\/script>/gs, "");
+        }
+    },
+    _addPlaceholder() {
+        const placeholderEl = document.createElement("div");
+        placeholderEl.classList.add("s_embed_code_placeholder", "alert", "alert-info", "pt16", "pb16");
+        placeholderEl.textContent = _t("Your Embed Code snippet doesn't have anything to display. Click on Edit to modify it.");
```

Comment:
Addition of translatable string in stable: you have to extract it in a new .pot (debug -> Settings -> Translation -> Export -> empty translation -> app per app).

Reply:
In general just add yours. Or you could make a PR about synchronizing the .pot first and we'll merge before this one.

Reply:
Note that if you synchronize the `.pot` somehow (here or in another PR), it's only about synchronizing the added terms, not the removed terms, so it's not just about replacing the file but just about adding the missing terms.

And if you just add the ones related to your PR, don't forget to put them exactly where it would be put if you generate the file: alphabetically.

Useful links:
https://github.com/odoo/odoo/pull/112416#discussion_r1173911412
https://discord.com/channels/803943906625716244/981917722826903572/1161603743599579207
https://github.com/odoo/odoo/pull/113570#pullrequestreview-1313445887

Reply:
> And if you just add the ones related to your PR, don't forget to put them exactly where it would be put if you generate the file: alphabetically.

Well obvisouly, never do it by hand, always replace the .pot file and commit the relavant parts, that way you control what you do and ensure it is at the right place / right format.